tests/functional/base.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (C) 2017  Custodia Project Contributors - see LICENSE file
+from __future__ import absolute_import
+
+import os
+import shutil
+import socket
+import subprocess
+import sys
+import time
+from string import Template
+
+import pytest
+
+from custodia.client import CustodiaHTTPClient
+from custodia.compat import url_escape
+from custodia.server.args import parse_args
+from custodia.server.config import parse_config
+
+
+class CustodiaServerRunner(object):
+    request_headers = {'REMOTE_USER': 'me'}
+    test_dir = 'tests/functional/tmp'
+    unique_number = 0
+
+    @classmethod
+    def setup_class(cls):
+        if os.path.isdir(cls.test_dir):
+            shutil.rmtree(cls.test_dir)
+        os.makedirs(cls.test_dir)
+
+    @classmethod
+    def teardown_class(cls):
+        shutil.rmtree(cls.test_dir)
+
+    def _wait_pid(self, process, wait):
+        timeout = time.time() + wait
+        while time.time() < timeout:
+            pid, status = os.waitpid(process.pid, os.WNOHANG)
+            if pid == process.pid:
+                return True
+            time.sleep(0.1)
+        return False
+
+    def _wait_socket(self, process, wait):
+        timeout = time.time() + wait
+        while time.time() < timeout:
+            if self.process.poll() is not None:
+                raise AssertionError(
+                    "Premature termination of Custodia server")
+            try:
+                s = socket.socket(family=socket.AF_UNIX)
+                s.connect(self.env['CUSTODIA_SOCKET'])
+            except OSError:
+                pass
+            else:
+                return True
+            time.sleep(0.1)
+        raise OSError('Timeout error')
+
+    def get_unique_number(self):
+        CustodiaServerRunner.unique_number = self.unique_number + 1
+        return CustodiaServerRunner.unique_number
+
+    @pytest.fixture(scope="class")
+    def simple_configuration(self):
+        with open('tests/functional/conf/template_simple.conf') as f:
+            configstr = f.read()
+
+        self.custodia_conf = os.path.join(self.test_dir, 'custodia.conf')
+        with (open(self.custodia_conf, 'w+')) as conffile:
+            t = Template(configstr)
+            conf = t.substitute({'TEST_DIR': self.test_dir})
+            conffile.write(conf)
+
+        self.args = parse_args([self.custodia_conf])
+        self.parser, self.config = parse_config(self.args)
```

Comment:
since you are not using the self.parser, would be better change this for:

`_ , self.config = ...`